registry/client/repository.go

```diff
@@ -604,6 +607,25 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo distribution.Repository, dgst digest.Digest) (distribution.Descriptor, error) {
```

Comment:
@aaronlehmann I see. I think the only reason this requires a fully formed `Repository` is on the registry side. I think we may be able to look it up inside of `linkedBlobStore`, so reducing this to a reference may be possible.

@BrianBland Thoughts?